sdk/eventhub/event-hubs/test/eventHubConsumerClient.spec.ts

```diff
@@ -225,6 +225,200 @@ describe("EventHubConsumerClient", () => {
       await producerClient.close();
     });
 
+    describe("Reinitialize partition processing after error", function(): void {
+      it("when subscribed to single partition", async function(): Promise<void> {
+        const partitionId = "0";
+        const consumerClient1 = new EventHubConsumerClient(
+          EventHubConsumerClient.defaultConsumerGroupName,
+          service.connectionString,
+          service.path
+        );
+        const consumerClient2 = new EventHubConsumerClient(
+          EventHubConsumerClient.defaultConsumerGroupName,
+          service.connectionString,
+          service.path
+        );
+
+        clients.push(consumerClient1, consumerClient2);
+
+        // keep track of the handlers called on subscription 1
+        const handlerCalls = {
+          initialize: 0,
+          close: 0
+        };
+        const subscriptionHandlers1: SubscriptionEventHandlers = {
+          async processError() {},
+          async processEvents() {
+            if (!handlerCalls.close) {
+              // start the 2nd subscription that will kick the 1st subscription off
+              subscription2 = consumerClient2.subscribe(partitionId, subscriptionHandlers2, {
+                ownerLevel: 1,
+                maxBatchSize: 1,
+                maxWaitTimeInSeconds: 1
+              });
+            } else {
+              // stop this subscription, we know close was called so we've restarted
+              await subscription1.close();
+            }
+          },
+          async processClose() {
+            handlerCalls.close++;
+          },
+          async processInitialize() {
+            handlerCalls.initialize++;
+          }
+        };
+        const subscriptionHandlers2: SubscriptionEventHandlers = {
+          async processError() {},
+          async processEvents() {
+            // stop this subscription since it already should have forced the 1st subscription to have an error.
+            await subscription2!.close();
+          }
+        };
+        let subscription2: Subscription | undefined;
+        const subscription1 = consumerClient1.subscribe(partitionId, subscriptionHandlers1, {
+          maxBatchSize: 1,
+          maxWaitTimeInSeconds: 1
+        });
+
+        await loopUntil({
+          maxTimes: 10,
+          name: "Wait for subscription1 to recover",
+          timeBetweenRunsMs: 5000,
+          async until() {
+            return !subscription1.isRunning && !subscription2!.isRunning;
+          }
+        });
+
+        handlerCalls.initialize.should.be.greaterThan(1);
```

Comment:
I think it's worth commenting here (if I understand this correctly) that the reason we don't know the exact number of times is because the first subscriber is just continually losing to the second subscriber (which has an owner level that's higher). So it'll just continue to try to reclaim the partition and continue to fail.

Reply:
Your understanding is correct, I'll add a comment.